src/opm/output/eclipse/AggregateGroupData.cpp

```diff
@@ -360,79 +325,26 @@ void productionGroup(const Opm::Schedule&     sched,
-         if (group control mode is NOT == ("FLD" OR "NONE"))
-            iGrp[nwgmax + 5] = -1
-         else
-             iGrp[nwgmax + 5] = 1
-     else if (group control mode is == ("FLD" OR "NONE"))
-            iGrp[nwgmax + 5] = -1
-         else
-             iGrp[nwgmax + 5] = -1
-
-    */
-    // default value
-
     const auto& cgroup = controlGroup(sched, sumState, group, simStep);
     const auto& deck_cmode = group.prod_cmode();
-    // Start branching for determining iGrp[nwgmax + 5]
-    // use default value if group is not available for group control
-
-    if (cgroup && cgroup->name() == "FIELD")
-        throw std::logic_error("Got cgroup == FIELD - uncertain logic");
 
+    if (cgroup && (group.getGroupType() != Opm::Group::GroupType::NONE)) {
+        auto cgroup_control = (cgroup->name() == "FIELD") ? static_cast<int>(sumState.get("FMCTP", 0)) : static_cast<int>(sumState.get_group_var(cgroup->name(), "GMCTP", 0));
+        iGrp[nwgmax + IGroup::ProdActiveCMode]
+            = (prod_guide_rate_def != Opm::Group::GuideRateProdTarget::NO_GUIDE_RATE) ? cgroup_control : 0;
+    } else {
+        iGrp[nwgmax + IGroup::ProdActiveCMode] = Opm::Group::ProductionCMode2Int(active_cmode);
 
-    iGrp[nwgmax + 5] = -1;
-    if (groupProductionControllable(sched, sumState, group, simStep)) {
-        // this section applies if group is controllable - i.e. has wells that may be controlled
-        if (!group.productionGroupControlAvailable() && (!cgroup)) {
-            // group can respond to higher level control
-            iGrp[nwgmax + 5] = 0;
-            goto CGROUP_DONE;
-        }
-
-
-        if (cgroup) {
-            iGrp[nwgmax + 5] = 1;
-            if (prod_guide_rate_def != Opm::Group::GuideRateProdTarget::NO_GUIDE_RATE) {
-                if (deck_cmode == Opm::Group::ProductionCMode::FLD)
-                    iGrp[nwgmax + 5] = cgroup->insert_index();
-
-                if (deck_cmode == Opm::Group::ProductionCMode::NONE)
-                    iGrp[nwgmax + 5] = cgroup->insert_index();
-            }
-            goto CGROUP_DONE;
-        }
-
-
-        if (higherLevelProdCMode_NotNoneFld(sched, group, simStep)) {
-
-            if (deck_cmode == Opm::Group::ProductionCMode::FLD)
-                iGrp[nwgmax] = 1;
-            if (deck_cmode == Opm::Group::ProductionCMode::NONE)
-                iGrp[nwgmax] = 1;
-
-            goto CGROUP_DONE;
-        }
-
-    } else if (deck_cmode == Opm::Group::ProductionCMode::NONE) {
-        iGrp[nwgmax + 5] = 1;
+        // The PRBL and CRAT modes where not handled in a previous explicit if
+        // statement; whether that was an oversight or a feature?
+        if (active_cmode == Opm::Group::ProductionCMode::PRBL || active_cmode == Opm::Group::ProductionCMode::CRAT)
+            iGrp[nwgmax + IGroup::ProdActiveCMode] = 0;
     }
- CGROUP_DONE:
+    iGrp[nwgmax + 9] = iGrp[nwgmax + IGroup::ProdActiveCMode];
```

Comment:
Do you have a name for item `nwgmax + 9`?

Reply:
Since this is currently interpreted as a copy of the IGroup::ProdActiveCMode item, it has not been given a separate name.